File: java/Day01.java
import java.util.Scanner;

public class Day01 {
    private static int move(char c) {
        switch (c) {
            case '(':
                return 1;
            case ')':
                return -1;
            default:
                return 0;
        }
    }

    private static int part1(String input) {
        int floor = 0;
        for (char c : input.toCharArray()) {
            floor += move(c);
        }
        return floor;
    }

    private static int part2(String input) {
        int floor = 0;
        int position = 0;
        while (true) {
            floor += move(input.charAt(position));
            position++;
            if (floor == -1)
                return position;
        }
    }

    public static void main(String[] args) {
        var scanner = new Scanner(System.in);
        var input = scanner.nextLine();

        System.out.println("Part 1: " + part1(input));
        System.out.println("Part 2: " + part2(input));

        scanner.close();
    }

}

File: kotlin/day_01.kt
fun move(input: Char): Int {
    when (input) {
        '(' -> return 1
        ')' -> return -1
    }
    return 0
}

fun part_1(input: String): Int {
    var floor = 0

    input.forEach { floor += move(it) }

    return floor
}

fun part_2(input: String): Int {
    var floor = 0
    var index = 0

    while (floor != -1) {
        floor += move(input[index])
        index++
    }

    return index
}

fun main() {
    var input = readln()

    println("Part 1: ${part_1(input)}")
    println("Part 2: ${part_2(input)}")
}
